realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -4560,6 +4560,58 @@ public void hittingMaxNumberOfVersionsThrows() {
         }
     }
 
+    @Test
+    @RunTestInLooperThread
+    // FIXME Maybe a bit convoluted, consider deleting it again.
+    public void numberOfVersionsDecreasedOnClose() {
+        int count = 50;
+        final CountDownLatch bgThreadDoneLatch = new CountDownLatch(count);
+
+        RealmConfiguration config = configFactory.createConfigurationBuilder()
+                // The multiple embedded threads seems to cause trouble with factory's directory setting
+                .directory(context.getFilesDir())
+                .name("versions-test.realm")
+                .maxNumberOfActiveVersions(5)
+                .build();
+
+        // Read only instance
+        looperThread.postRunnable(() -> {
+            Realm realm = Realm.getInstance(config);
+            realm.addChangeListener(realm1 -> {
+                // FIXME Seems like versions are not deprecated if we do queries here
+                //int objectCount = realm.where(AllJavaTypes.class).findAll().size();
```

Comment:
Replacing this with `long objectCount = realm.where(AllJavaTypes.class).count();` works. This could indicate that it is the TableView created by `findAll()` that somehow isn't being updated correctly thus leaking the version and causing the test to crash.

I'll try to move this to ObjectStore and see if I can replicate the behavior there.